### PR TITLE
admin product list

### PR DESCRIPTION
**Pull request by ghost**
## Related Tickets
- ticket redmine: https://edu-redmine.sun-asterisk.vn/issues/76108

## WHAT (optional)
- Admin view product list

## Evidence (Screenshot or Video)
image

### CODE REVIEW

**Comment by ghost** [comment]
ready

File: src/controllers/admin/product.controller.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  private readonly productService: ProductService;
+  constructor() {
+    this.productService = new ProductService();
+  }
+
+  public getAdminProductPage = asyncHandler(
+    async (req: Request, res: Response) => {
+      const admin = (req.session as CustomSessionData).admin;
+      const pageOptions = plainToClass(ProductPageOptions, req.query);
+      console.log('pageOptions:', pageOptions);
```
Comment (huongnt-2545) [inline src/controllers/admin/product.controller.ts]: xóa debug đi e nha

File: src/controllers/admin/product.controller.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+          currentSite: AdminNavBar.PRODUCT,
+          categories,
+          errors,
+          query: pageOptions,
+          formatDate,
+          formatCurrency,
+        });
+      }
+      const productPage = await this.productService.getProductPage(pageOptions);
+      console.log('Products:', productPage.data);
```
Comment (huongnt-2545) [inline src/controllers/admin/product.controller.ts]: xóa debug

File: src/controllers/product.controller.ts [hunk 3]
```diff
@@ -64,6 +64,7 @@ export class ProductController {
     async (req: Request, res: Response) => {
       const user = (req.session as CustomSessionData).user;
       const pageOptions = plainToClass(ProductPageOptions, req.query);
+      console.log('Page options: ', pageOptions);
```
Comment (huongnt-2545) [inline src/controllers/product.controller.ts]: xóa debug

File: src/views/admin/product/list.ejs [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+                                    </button>
+                                  </li>
+                                  <% } %>
+                                    <% if (meta.hasNextPage) { %>
+                                      <li class="page-item">
+                                        <button type="submit" class="page-link" name="page" value="<%= parseInt(meta.page) + 1 %>">
+                                          <%=t('page.next')%>
+                                        </button>
+                                      </li>
+                                      <% } %>
```
Comment (huongnt-2545) [inline src/views/admin/product/list.ejs]: e check lại dent đoạn này nha

**Comment by ghost** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]

**Pull request merged**